finalitygadget/finalitygadget.go

```diff
@@ -207,8 +210,38 @@ func (fg *FinalityGadget) QueryIsBlockBabylonFinalized(block *types.Block) (bool
 	return true, nil
 }
 
-/* QueryBlockRangeBabylonFinalized searches for a row of consecutive finalized blocks in the block range, and returns
- * the last finalized block height
+// QueryIsBlockBabylonFinalized queries the finality status of a given block height from the internal db
+func (fg *FinalityGadget) QueryIsBlockBabylonFinalized(block *types.Block) (bool, error) {
+	// check if the finality gadget is enabled
+	// if not, always return true to pass through op derivation pipeline
+	isEnabled, err := fg.cwClient.QueryIsEnabled()
+	if err != nil {
+		return false, err
+	}
+	if !isEnabled {
+		return true, nil
+	}
+
+	// convert the L2 timestamp to BTC height
+	btcblockHeight, err := fg.btcClient.GetBlockHeightByTimestamp(block.BlockTimestamp)
+	if err != nil {
+		return false, err
+	}
+
+	// check whether the btc staking is activated
+	btcStakingActivatedTimestamp, err := fg.QueryBtcStakingActivatedTimestamp()
+	if err != nil {
+		return false, err
+	}
+	if btcblockHeight < btcStakingActivatedTimestamp {
+		return false, types.ErrBtcStakingNotActivated
+	}
```

Comment:
i think there is no need to check this. we only insert blocks to the db after BTC finalization.

this function should simply read from DB

it's also cleaner so we don't need to copy the above code

Reply:
we need these checks because the function can be called by op node before btc staking is activated. It should return the correct error `ErrBtcStakingNotActivated` if btc staking is not activated

if we just return the db call, the error type will be incorrect (i think one of the e2e tests broke for this reason)

Reply:
ah so `QueryIsBlockBabylonFinalized` is not just called by `processHeight`?

Reply:
yep, so:
- `QueryIsBlockBabylonFinalized` is external
- `QueryIsBlockBabylonFinalizedFromBabylon` will eventually be internal (rename `queryIsBlockBabylonFinalizedFromBabylon` once fully tested)